File: 8441-1184-735-asteroid-collision/8441-1184-735-asteroid-collision.java
class Solution {
    public int[] asteroidCollision(int[] asteroids) {
        Stack<Integer> s = new Stack<>();

        for(int a : asteroids){
            while(!s.empty() && s.peek()>0 && a<0){
                if(s.peek()<Math.abs(a))
                    s.pop();
                else{
                    if(s.peek()==Math.abs(a))
                        s.pop();
                    a = 0;
                    break;
                }
            }
            if(a!=0)
                s.push(a);
        }
        int a[] = new int[s.size()];
        for(int i=s.size()-1;i>-1;i--)
            a[i] = s.pop();
        return a;
    }
}

File: 8441-1184-735-asteroid-collision/8441-1184-735-asteroid-collision.js
/**
 * @param {number[]} asteroids
 * @return {number[]}
 */
var asteroidCollision = function(a) {
    const s = []
    
    for(let i=0;i<a.length;i++){
        let flag = false
        while(s.length>0 && s[s.length-1]>0 && a[i]<0){
            let t = s[s.length-1]
            if(t<Math.abs(a[i]))
                s.pop()
            else if(t===Math.abs(a[i])){
                s.pop()
                a[i]=0
                break
            }else{
                a[i]=0
                break
            }
        }
        if(a[i]!==0)
            s.push(a[i])
    }
    return s
};